frontend/packages/kubevirt-plugin/src/components/vms/VMDetailsItem.tsx

```diff
@@ -0,0 +1,65 @@
+import { Button } from '@patternfly/react-core';
+import { LoadingInline } from '@console/internal/components/utils';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { EditButton } from '../edit-button';
+
+const VMDetailsItem: React.FC<VMDetailsItemProps> = ({
+  title,
+  canEdit = false,
+  editButtonId,
+  onEditClick,
+  idValue,
+  isLoading = false,
+  isNotAvail = false,
+  isNotAvailMessage,
+  valueClassName,
+  arePendingChanges,
+  children,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <>
+      <dt>
+        <span>
+          {title} <EditButton id={editButtonId} canEdit={canEdit} onClick={onEditClick} />
+          {arePendingChanges && (
+            <Button
+              className="co-modal-btn-link--inline"
+              variant="link"
+              isInline
+              onClick={onEditClick}
+            >
+              {t('kubevirt-plugin~View Pending Changes')}
+            </Button>
+          )}
+        </span>
+      </dt>
+      <dd id={idValue} className={valueClassName}>
+        {isNotAvail && (
+          <span className="text-secondary">
+            {isNotAvailMessage || t('kubevirt-plugin~Not available')}
+          </span>
+        )}
+        {isLoading && !isNotAvail && <LoadingInline />}
+        {!isNotAvail && !isLoading && children}
+      </dd>
```

Comment:
I find these conditions hard to follow. Can you do something like

```suggestion
if (isNotAvail) {
  body = (
    <span className="text-secondary">
      {isNotAvailMessage || t('kubevirt-plugin~Not available')}
    </span>
  );
} else if (isLoading) {
  body = <LoadingInline />;
} else {
  body = children;
}



....
      <dd id={idValue} className={valueClassName}>
         {body}
      </dd>
```